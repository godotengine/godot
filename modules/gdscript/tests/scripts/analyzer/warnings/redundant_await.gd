signal my_signal()

# CI cannot test async things.
func test_signals():
	await my_signal
	var t: Signal = my_signal
	await t

func coroutine() -> void:
	@warning_ignore("redundant_await")
	await 0

func not_coroutine_variant():
	pass

func not_coroutine_void() -> void:
	pass

func test():
	const CONST_NULL = null
	var var_null = null
	var var_int: int = 1
	var var_variant: Variant = 1
	var var_array: Array = [1]

	await CONST_NULL
	await var_null
	await var_int
	await var_variant
	await var_array[0]

	await coroutine
	await coroutine()
	await coroutine.call()
	await self.coroutine()
	await call(&"coroutine")

	await not_coroutine_variant
	await not_coroutine_variant()
	await self.not_coroutine_variant()
	await not_coroutine_variant.call()
	await call(&"not_coroutine_variant")

	await not_coroutine_void
	await not_coroutine_void()
	await self.not_coroutine_void()
	await not_coroutine_void.call()
	await call(&"not_coroutine_void")

	var callable: Callable = coroutine
	await callable
	await callable.call()
	await callable.get_method()
