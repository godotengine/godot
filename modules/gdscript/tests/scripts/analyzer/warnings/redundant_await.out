GDTEST_OK
~~ WARNING at line 26: (REDUNDANT_AWAIT) "await" keyword not needed in this case, because the expression isn't a coroutine nor a signal.
~~ WARNING at line 28: (REDUNDANT_AWAIT) "await" keyword not needed in this case, because the expression isn't a coroutine nor a signal.
~~ WARNING at line 32: (REDUNDANT_AWAIT) "await" keyword not needed in this case, because the expression isn't a coroutine nor a signal.
~~ WARNING at line 38: (REDUNDANT_AWAIT) "await" keyword not needed in this case, because the expression isn't a coroutine nor a signal.
~~ WARNING at line 44: (REDUNDANT_AWAIT) "await" keyword not needed in this case, because the expression isn't a coroutine nor a signal.
~~ WARNING at line 45: (REDUNDANT_AWAIT) "await" keyword not needed in this case, because the expression isn't a coroutine nor a signal.
~~ WARNING at line 46: (REDUNDANT_AWAIT) "await" keyword not needed in this case, because the expression isn't a coroutine nor a signal.
~~ WARNING at line 51: (REDUNDANT_AWAIT) "await" keyword not needed in this case, because the expression isn't a coroutine nor a signal.
~~ WARNING at line 53: (REDUNDANT_AWAIT) "await" keyword not needed in this case, because the expression isn't a coroutine nor a signal.
